ext/tiny_tds/client.c

```diff
@@ -57,6 +57,15 @@ VALUE rb_tinytds_raise_error(DBPROCESS *dbproc, int is_message, int cancel, cons
 
 
 // Lib Backend (Memory Management & Handlers)
+static void push_userdata_error(tinytds_client_userdata *userdata, tinytds_errordata error) {
+  // reallocate memory for the array as needed
+  if (userdata->nonblocking_errors_size == userdata->nonblocking_errors_length) {
+    userdata->nonblocking_errors_size *= 2;
+    userdata->nonblocking_errors = realloc(userdata->nonblocking_errors, userdata->nonblocking_errors_size * sizeof(tinytds_errordata));
+  }
+
+  userdata->nonblocking_errors[userdata->nonblocking_errors_length++] = error;
```

Comment:
This method might get marginally simpler if `userdata->nonblocking_errors_length += 1` is added to the top of the method, then subsequent sizes and so forth are compared to that.

Reply:
I opted for incrementing at the end to prevent the need for `userdata->nonblocking_errors_length - 1` in other places. 

```suggestion
  userdata->nonblocking_errors[userdata->nonblocking_errors_length] = error;
  userdata->nonblocking_errors_length++;
```